core/src/main/scala/org/apache/spark/h2o/backends/external/RestApiUtils.scala

```diff
@@ -38,6 +38,39 @@ trait RestApiUtils extends RestCommunication {
     hc.getOrElse(H2OContext.ensure()).getConf.get("spark.ext.h2o.rest.api.based.client", "false") == "true"
   }
 
+  def convertAllStringVecToCategorical(conf: H2OConf, frameId: String): H2OFrame = {
```

Comment:
Maybe these methods could be extracted to a decicate class/trait

Reply:
Yup, agree, I think it is time that we split the RestAPI Utils to several helper classes or traits

Reply:
I was thinking about it and I actually think that we should put these classes in the new package as well, something like`ai.h2o.sparkling.rest` and then create classes like `Frame`, `Model` etc
I have created JIRA for this task https://0xdata.atlassian.net/browse/SW-1937 ( I would like to do the split and moving after we merge your change to avoid unnecessary rebase conflicts)

I will still move these new methods to the new package. Thanks!

Reply:
Done 👍

Reply:
I have made the frame relates classes available on `ai.h2o.sparkling.H2OFrame` so we can slowly start creating our Scala H2OFrame API via Rest

Reply:
That sounds great!